common/src/main/java/org/conscrypt/NativeCrypto.java

```diff
@@ -1443,6 +1443,11 @@ static native void ENGINE_SSL_force_read(long ssl, NativeSsl ssl_holder,
     static native void ENGINE_SSL_shutdown(long ssl, NativeSsl ssl_holder, SSLHandshakeCallbacks shc)
             throws IOException;
 
+    /**
+     * Generates a key from a password and salt using Scrypt.
+     */
+    static native byte[] Scrypt_generate_key(byte[] password, byte[] salt, long n, long r, long p, int key_len);
```

Comment:
We should make these ints here and in the JNI layer now that we changed the params.,  but I can do that as a followup.